core/script_language.h

```diff
@@ -35,6 +35,7 @@
 #include "core/map.h"
 #include "core/pair.h"
 #include "core/resource.h"
+#include "editor/doc_data.h"
```

Comment:
`core/` should not have dependencies on `editor/` code, so we need to find a different solution here. (At the very list in its current state this should be within `#ifdef TOOLS_ENABLED`, but I think we should find a solution that doesn't break encapsulation instead.)

CC @reduz

Reply:
A solution might be to move `editor/doc_data.h` to `core/doc_data.h` (to be discussed with @reduz).

Reply:
13:30 <Akien> reduz: Can you check my review comments at the end of #41095 ? It's pretty solid work from thakeenathees[m] and already approved by vnen, but I have some concerns on the use of `editor/doc_data.h` in `core/script_language.h`
13:30 <Akien> And overall there's a lot of `#ifdef TOOLS_ENABLED` code added in GDScript too which can lead to bugs in tools vs non-tools builds, but I don't really see a better way around it.
13:31 <Akien> I guess doc_data.h should just be moved to core?
13:32 <reduz> I am not sure
13:33 <reduz> it needs scene stuff too
13:33 <reduz> so probably no
13:33 <reduz> maybe we can create an intermediate class with just the structures
13:33 <reduz> and put it in core
13:35 <reduz> another alternative can be to just replicate some of the structures in ScriptLanguage
13:35 <reduz> with an ifdef tools
13:35 <reduz> but it looks like they are the same thing
13:35 <Akien> Seems it only needs scene for `scene/resources/theme.h`, but yeah that's true.
13:35 <reduz> I think best would be to just split it in two
13:35 <reduz> not sure what vnen thinks
13:36 <Akien> Yeah duplicating the struct might be the simplest, only ClassDoc is needed it seems.
13:36 <Akien> (Well ClassDoc depends on all other structs :))
13:36 <reduz> ClassDoc includes all the other structs
13:36 <reduz> yeah, I think the best is to split it
13:38 <Akien> So `core/doc_data.h` with the structs, and `editor/doc_manager.h` with the methods or something like that?
13:39 <Akien> Or DocTools
13:39 <reduz> yeah
```

Reply:
This class should be split in two, so you can access doc without having to rely on editor stuff.

Reply:
since doc_data.h, isn't depend on the editor, I've moved it to the core `core/doc_data.h` and changed all `#include "editor/doc_data.h"` into `#include "core/doc_data.h"` and it's working just fine and I didn't need to split it to an interface and implementation between core and editor, is it okey?
@akien-mga  @reduz

Reply:
The problem is that `doc_data.cpp` depends on `scene`, and `core` code should not depend on `scene`.

But you can have the data structs in `core` as `DocData`, and the actual utility methods for handling doc data in `editor` as e.g. `DocManager` or `DocTool` (which includes `core/doc_data.h`).

So it would be something like:
```
core/doc_data.h
editor/doc_tools.cpp
editor/doc_tools.h
```

Reply:
done! I've created a new commit for the doc_data refactor.